### ISSUE TITLE
zcash_client_backend: Use consistent ordering for type parameters in `scanning` and `wallet` types.

### ISSUE THREAD

**Issue by nuttycom** [zcash/librustzcash#1204]
@str4d requested a number of changes similar to the following in #1202; these are being deferred until after #1195 to avoid severe rebase issues.

> Minor nit: `vks` is passed in `(A, K)` order, so I'd prefer the generic parameters to be in that order as well.

_Originally posted by @str4d in https://github.com/zcash/librustzcash/pull/1202#discussion_r1506782790_
